lib/resources/elasticsearch.rb

```diff
@@ -0,0 +1,151 @@
+          .add(:cluster_name,           field: 'cluster_name')
+          .add(:name,                   field: 'name')
+          .add(:transport_address,      field: 'transport_address')
+          .add(:host,                   field: 'host')
+          .add(:ip,                     field: 'ip')
+          .add(:version,                field: 'version')
+          .add(:build_hash,             field: 'build_hash')
+          .add(:total_indexing_buffer,  field: 'total_indexing_buffer')
+          .add(:roles,                  field: 'roles')
+          .add(:settings,               field: 'settings')
+          .add(:os,                     field: 'os')
+          .add(:process,                field: 'process')
+          .add(:jvm,                    field: 'jvm')
+          .add(:transport,              field: 'transport')
+          .add(:http,                   field: 'http')
+          .add(:plugins,                field: 'plugins')
+          .add(:plugin_list,            field: 'plugin_list')
+          .add(:modules,                field: 'modules')
+          .add(:module_list,            field: 'module_list')
+          .add(:node_id,                field: 'node_id')
+          .add(:ingest,                 field: 'ingest')
+          .add(:exists?) { |x| !x.entries.empty? }
+          .add(:count) { |t, _|
+            t.entries.length
+          }
+
+    filter.connect(self, :nodes)
+
+    attr_reader :nodes
+    def initialize(nodes)
+      @nodes = nodes
+    end
+  end
+
+  class Elasticsearch < Inspec.resource(1)
+    name 'elasticsearch'
+    desc "Use the Elasticsearch InSpec audit resource to test the Elasticsearch
+      configuration settings. The data is extract from the live elastic search
+      cluster using the curl command `curl localhost:9200/_nodes/`. The resource
+      makes uses of FilterTable and its functionality can be used "
+
+    example "
+      describe elasticsearch.nodes do
+        it { should exist }
+      end
+      describe elasticsearch.nodes do
+        its('name') { should include 'my_node' }
+      end
+      describe elasticsearch.nodes do
+        its('name') { should_not include 'node_name' }
+      end
+      elasticsearch.nodes.os.each do |nodeos|
+        describe nodeos do
+          its('name'){ should_not cmp 'MacOS' }
+        end
+      end
+      elasticsearch.nodes.version.each do |node_version|
+        describe node_version do
+          it{ should be > '1.2.0' }
+        end
+      end
+      elasticsearch.nodes.where{os.name == 'Linux' }.settings.each do |node_settings|
+        describe node_settings do
+          its('path.conf') {should cmp '/etc/elasticsearch'}
+        end
+      end
+      elasticsearch.nodes.where{name == 'package-centos-72' }.settings.each do |node_settings|
+        describe node_settings do
+          its('path.conf') {should cmp '/etc/elasticsearch'}
+        end
+      end
+        "
+
+    def initialize(url = 'http://localhost:9200/_nodes/')
```

Comment:
Any reason to include the `_nodes` part here? I think perhaps this should be the base URL for wherever Elasticsearch lives, and then we can call `/_nodes` within the resource. That way if we want to hit other endpoints later, such as `/_cluster/health`, we don't have to rework this.

Reply:
The _nodes part is important because it exposes all the settings we are exposing using the resource and our parsers are expecting the json structure the '_nodes' curl request returns.

Reply:
@rx294 that's exactly my point. Your resource depends on hitting `/_nodes` to do its job, but that is not the top-level URL to an Elasticsearch node/cluster. You are shifting an implementation detail responsibility onto the consumer of your resource.

You should ask expect the user to give you the top-level URL to their cluster, and then **you** as the resource author should tack on the `/_nodes` endpoint to the URL before you start making curl calls. Otherwise, you are forcing users to know that detail... and in the future, if we need to retrieve data from different endpoints (such as `/_cluster` or `/_cat`, we don't have to do janky regsub or break backward compatibility.